### PR TITLE
[ONNX] Update ONNX doc for writing pytorch model

### PR DESCRIPTION
**Pull request by jiafatom**
For tracing successfully, we need write pytorch model in torch way. So we add instructions with examples here.

### CODE REVIEW

File: docs/source/onnx.rst [hunk 1]
```diff
@@ -249,6 +249,31 @@ E.g.: ::
...
+For example, do not use numpy operators on numpy tensors: ::
+
+    np.concatenate((x, y, z), axis=1)
+
+do not convert to numpy types: ::
+
+    y = x.astype(np.int)
+
+Always use torch tensors and torch operators: torch.concat, etc.
+In addition, Dropout layer need defined in init function so that inferencing can handle it properly, i.e., ::
```
Comment (KsenijaS) [inline docs/source/onnx.rst:268]: Does this only apply to Dropout, what about Batchnorm, or any other op that operates differently in training vs inferencing?

Comment (jiafatom) [reply docs/source/onnx.rst:268]: Not sure about Batchnorm. The difference is that Dropout in inferencing will be Identity, but Batchnorm is Batchnorm for both training/inferencing case. This may cause the difference. So far we only meet the Dropout layer issue, so we just include Dropout here. We can add more if we further meet new issues.

File: docs/source/onnx.rst [hunk 2]
```diff
@@ -249,6 +249,31 @@ E.g.: ::
     out = model(*inputs)
     torch.onnx.export(model, inputs, 'loop_and_list.onnx', opset_version=11, example_outputs=out)
 
+Write pytorch model in Torch way
```
Comment (BowenBao) [inline docs/source/onnx.rst:252]:
```suggestion
Write PyTorch model in Torch way
```

File: docs/source/onnx.rst [hunk 3]
```diff
@@ -249,6 +249,31 @@ E.g.: ::
...
+--------------------------------
+
+Pytorch models can be written using numpy manipulations, but this is not proper when we convert to the onnx model.
```
Comment (BowenBao) [inline docs/source/onnx.rst:255]:
```suggestion
PyTorch models can be written using numpy manipulations, but this is not proper when we convert to the ONNX model.
```

File: docs/source/onnx.rst [hunk 4]
```diff
@@ -249,6 +249,31 @@ E.g.: ::
...
+For the trace-based exporter, tracing treats the numpy values as the constant node,
+therefore it calculates the wrong result if we change the input.
+So the pytorch model need implement using torch operators.
```
Comment (BowenBao) [inline docs/source/onnx.rst:258]:
```suggestion
So the PyTorch model need implement using torch operators.
```

**Review by BowenBao — APPROVED** [review]
LGTM, minor comments on capitalization. Thanks.

File: docs/source/onnx.rst (hunk 2 above)
Comment (jiafatom) [reply docs/source/onnx.rst:252]: done

**Comment by jiafatom** [comment]
@bzinodev, could you please take a look? thanks.

**Review by bzinodev — APPROVED** [review]
